vulnerabilities/scraper/npm.py

```diff
@@ -113,12 +107,15 @@ def scrape_vulnerabilities():
     """
     Extract JSON From NPM registry
     """
-    nextpage = PAGE
     package_vulnerabilities = []
-    while nextpage:
-        cururl = NPM_URL.format(nextpage)
-        response = json.load(urlopen(cururl))
-        package_vulnerabilities.extend(extract_data(response))
-        next_page = response.get('urls', {}).get('next')
+    for page_num in count():
```

Comment:
As I mentioned [on the ticket](https://github.com/nexB/vulnerablecode/issues/162#issuecomment-599063793), I don't think the error you encountered is fixed by this change. It was just a temporary issue of the registry backend.
IMHO the previous approach of following the URL to the next page included in the JSON response is preferrable so I propose we stick with that.

Reply:
Turns out it was not an issue of the registry and there has been a bug in the NPM importer all along. I would still like to stick with the current approach and just fix the bug: rename `next_page` to `nextpage` (or the other way around).

Reply:
It's not the only issue being fixed ,  for eg  if I change `next_page`   to `nextpage` (that was my initial approach btw)  one of the issue was this

```
>>> npm.scrape_vulnerabilities()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/workspace/vulnerablecode/vulnerabilities/scraper/npm.py", line 122, in scrape_vulnerabilities
    package_vulnerabilities.extend(extract_data(response))
  File "/workspace/vulnerablecode/vulnerabilities/scraper/npm.py", line 97, in extract_data
    obj.get('patched_versions', '')
  File "/workspace/vulnerablecode/vulnerabilities/scraper/npm.py", line 67, in extract_versions
    all_ver = get_all_versions(package_name)
  File "/workspace/vulnerablecode/vulnerabilities/scraper/npm.py", line 55, in get_all_versions
    with urlopen(package_url) as response:
  File "/home/gitpod/.pyenv/versions/3.7.6/lib/python3.7/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/home/gitpod/.pyenv/versions/3.7.6/lib/python3.7/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/home/gitpod/.pyenv/versions/3.7.6/lib/python3.7/urllib/request.py", line 641, in http_response
    'http', request, response, code, msg, hdrs)
  File "/home/gitpod/.pyenv/versions/3.7.6/lib/python3.7/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/home/gitpod/.pyenv/versions/3.7.6/lib/python3.7/urllib/request.py", line 503, in _call_chain
    result = func(*args)
  File "/home/gitpod/.pyenv/versions/3.7.6/lib/python3.7/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: Not Found
```
This is due to a package called 'panellum' whose data the registry doesnt have.
It's located at  https://registry.npmjs.org/-/npm/v1/security/advisories?page=127

Reply:
But that issue has a different cause and is handled in `get_all_versions()`.

I do want to merge that fix and the change to using `RangeSpecifier`. All I'm asking is that you fix this specific bug that caused an endless loop in a different way. The argument for that is, since the response from NPM includes a URL to the next page we should use that instead of making assumptions about the URL (even if they seem obviously correct).

Reply:
Yeah, that approach is better